go.mod

```diff
@@ -4,12 +4,14 @@ go 1.15
 
 require (
 	cloud.google.com/go v0.65.0
+	github.com/Azure/azure-amqp-common-go v1.1.4
```

Comment:
I don't know the details or anything, but I am curious, can't we reused v3 already imported on the line below?

Reply:
The package we are using for event hub is `github.com/Azure/azure-event-hubs-go v1.3.1` which depends on `github.com/Azure/azure-amqp-common-go v1.1.4`. This was hidden because we didn't need to directly reference any entities from the `aad` subcomponent directly.

But with this PR we are using the `aad.NewJWTProvider` directly because we are instantiating the hub by supplying the provider `eventhub.NewHub(info.Namespace, info.EventHubName, provider)`.

So it's either we keep this old (new :P) package reference, or we upgrade `azure_eventhub.go` to use the latest version of `github.com/Azure/azure-event-hubs-go`.

Will be happy to take either way.

Reply:
Thanks for the info, it would be great if you could do the upgrade of `azure_eventhub.go` as part of this PR. I think it is much better then keeping on this old dependency.

Reply:
Done